pkg/authz/auth.go

```diff
@@ -61,8 +63,27 @@ type ResourceAttributes struct {
 	Name        string `json:"name,omitempty"`
 }
 
-// NewAuthorizer creates an authorizer compatible with the kubelet's needs
-func NewAuthorizer(client authorizationclient.SubjectAccessReviewInterface) (authorizer.Authorizer, error) {
+// StaticAuthorizationConfig describes what is needed to specify a static
+// authorization.
+type StaticAuthorizationConfig struct {
+	User            UserConfig
+	Verb            string `json:"verb,omitempty"`
+	Namespace       string `json:"namespace,omitempty"`
+	APIGroup        string `json:"apiGroup,omitempty"`
+	Resource        string `json:"resource,omitempty"`
+	Subresource     string `json:"subresource,omitempty"`
+	Name            string `json:"name,omitempty"`
+	ResourceRequest *bool  `json:"resourceRequest,omitempty"`
```

Comment:
any reason this is a pointer? i think tristate logic is not really necessary here, either it is a resource request or not.

Reply:
> either it is a resource request or not.

Yes, the intention is to have the third state as "don't care". So user can configure one static auth that allows both resource and non-resource requests.
Not sure whether that is a valid scenario, say "allow all GET requests for a user" but figured going with tri-state wouldn't hurt anything.

Reply:
> allow all GET requests for a user

this is a non-resource authorization config (aka `resourceRequest = false`), hence my suggestion not to have tristate, which might be confusing :thinking:

Reply:
You are certainly the expert, my thinking was that a GET could potentially be a resource or non-resource. Like in your example here https://github.com/brancz/kube-rbac-proxy/pull/116#discussion_r622903161

Reply:
yeah, in case of GET we always are in "non-resource" mode (aka `resourceRequest = false`), so I think we can safely factor the tristate logic to a pure `true/false` one :+1: